Source/Client.swift

```diff
@@ -106,6 +107,7 @@ extension Client {
             )
 
             try connection.open(timingOut: now() + connectionTimeout)
```

Comment:
I think we should only try the connection when we're recreating one. When the connection already exists we shouldn't open it again.